src/hls-adapter.js

```diff
@@ -449,50 +450,117 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
 
   /**
    * Handles hls errors.
-   * @param {string} event - The event name.
+   * @param {number} severity - The error severity.
+   * @param {number} category - The error category.
+   * @param {number} code - The error code.
+   * @param {any} description - The error description.
+   * @param {any} callback - function to execute after throwing an error.
+   * @private
+   * @returns {void}
+   */
+  _handleFatalError(severity: number, category: number, code: number, description: any, callback: any): void{
+    const message = PlayerError.createError({
+      severity: severity,
+      category: category,
+      code: code,
+      args: {data: description}
+    });
+    this._trigger(BaseMediaSourceAdapter.CustomEvents.ERROR,message);
+    HlsAdapter._logger.error(description);
+    callback && callback();
```

Comment:
No reason to mix the error dispatching and error handing, better handle the "callback" logic where it is already handled now

Reply:
K